packages/core/src/internal/module-builder.ts

```diff
@@ -610,6 +614,68 @@ class IgnitionModuleBuilderImplementation<
     return future;
   }
 
+  public encodeFunctionCall<
+    ContractNameT extends string,
+    FunctionNameT extends string
+  >(
+    contractFuture: CallableContractFuture<ContractNameT>,
+    functionName: FunctionNameT,
+    args: ArgumentType[] = [],
+    options: EncodeFunctionCallOptions = {}
+  ): EncodeFunctionCallFuture<ContractNameT, FunctionNameT> {
+    if (!Array.isArray(args)) {
+      this._throwErrorWithStackTrace(
+        `Invalid parameter "args" provided to encodeFunctionCall "${functionName}" in module "${this._module.id}"`,
+        this.encodeFunctionCall
+      );
+    }
+
+    if (typeof options !== "object") {
+      this._throwErrorWithStackTrace(
+        `Invalid parameter "options" provided to encodeFunctionCall "${functionName}" in module "${this._module.id}"`,
+        this.encodeFunctionCall
+      );
+    }
+
+    const futureId = toCallFutureId(
```

Comment:
I am not sure about this. It is not obvious that this is not a call but the encoding of a call:

![image](https://github.com/NomicFoundation/hardhat-ignition/assets/24030/1984df3b-95b1-4784-a4db-246fd906a54d)

Maybe something like `Apollo#Rocket.encode[launch]`. I mean even writing it I hate it.

Reply:
I think not clashing it would be better. What about something like `Apollo#encodeFunctionCall(<functionCallId>)`?

Reply:
Should the underlying contract name be included as well? Is `functionCallId`:
1. `launch`
2. `Rocket.launch` or using some other separator?